core/src/main/java/org/fao/geonet/web/LocaleRedirects.java

```diff
@@ -77,8 +78,19 @@ public ModelAndView login(final HttpServletRequest request,
                               @RequestParam(value = NODE_PARAMETER, required = false) String node,
                               @CookieValue(value = Jeeves.LANG_COOKIE, required = false) String langCookie,
                               @RequestHeader(value = ACCEPT_LANGUAGE_HEADER, required = false) final String langHeader) {
-        String lang = lang(langParam, langCookie, langHeader);
-        return redirectURL(createServiceUrl(request, "catalog.signin", lang, node));
+        StringBuilder redirectURL;
+        try {
+            DefaultSavedRequest originalRequest = (DefaultSavedRequest) request.getSession().getAttribute("SPRING_SECURITY_SAVED_REQUEST");
+            redirectURL = new StringBuilder();
+            redirectURL.append(originalRequest.getContextPath() + originalRequest.getServletPath() + originalRequest.getPathInfo());
+            redirectURL.append("?login");
+            if (originalRequest.getQueryString() != null && originalRequest.getQueryString().length() > 0) {
```

Comment:
I fear we have to deal with `# ` too, eg with /geonetwork/srv/fre/catalog.search;jsessionid=15j8ejcv4twxuauduhjwsk9z4#/metadata/3ca21b5a-9c82-401a-81d8-308763e9fd2a

Reply:
I tested an URL with "#...." and this part of URL is not sent to the server.